From b943ed5d5b7fb53490a166e028fd8e76764ace0f Mon Sep 17 00:00:00 2001
Message-Id: <patch-1.3-b943ed5d5b7-20220406T104134Z-avarab@gmail.com>
In-Reply-To: <cover-0.3-00000000000-20220406T104134Z-avarab@gmail.com>
References: <Ybtb6Shdj56ACdub@coredump.intra.peff.net>
	<cover-0.3-00000000000-20220406T104134Z-avarab@gmail.com>
From: =?UTF-8?q?=C3=86var=20Arnfj=C3=B6r=C3=B0=20Bjarmason?=
 <avarab@gmail.com>
Date: Thu, 16 Dec 2021 13:00:54 +0100
Subject: [PATCH 1/3] cocci: add a rename of "struct child_process"'s
 "env_array" to "env"
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Start following-up on the rename mentioned in c7c4bdeccf3 (run-command
API: remove "env" member, always use "env_array", 2021-11-25) of
"env_array" to "env".

The "env_array" name was picked in 19a583dc39e (run-command: add
env_array, an optional argv_array for env, 2014-10-19) because "env"
was taken. Let's not forever keep the oddity of "*_array" for this
"struct strvec", but not for its "args" sibling.

This is added as a "pending" migration because it'll need a couple of
manual changes to go along with it, those will be performed in the
subsequent commit.

Signed-off-by: Ævar Arnfjörð Bjarmason <avarab@gmail.com>
---
 contrib/coccinelle/run_command.pending.cocci | 11 +++++++++++
 1 file changed, 11 insertions(+)
 create mode 100644 contrib/coccinelle/run_command.pending.cocci

diff --git a/contrib/coccinelle/run_command.pending.cocci b/contrib/coccinelle/run_command.pending.cocci
new file mode 100644
index 00000000000..709fdfcf720
--- /dev/null
+++ b/contrib/coccinelle/run_command.pending.cocci
@@ -0,0 +1,11 @@
+@@
+struct child_process E;
+@@
+- E.env_array
++ E.env
+
+@@
+struct child_process *E;
+@@
+- E->env_array
++ E->env
-- 
2.36.0.rc0.849.g2d5b5d9ab01

